lib/assertions.js

```diff
@@ -700,7 +700,11 @@ module.exports = function (expect) {
         '<object> to be (a map|a hash|an object) whose values [exhaustively] satisfy <assertion>'
     ], function (expect, subject, nextArg) {
         expect.errorMode = 'nested';
-        expect(subject, 'not to equal', {});
+        if (expect.subjectType.is('array')) {
```

Comment:
Yeah, that also seems to be the conclusion from the discussion we had at https://github.com/unexpectedjs/unexpected/pull/338

However, I think we should leave that as a separate discussion. I've opened https://github.com/unexpectedjs/unexpected/issues/350